MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -4171,3 +4165,13 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 uint64_t mvkGetRegistryID(id<MTLDevice> mtlDevice) {
 	return [mtlDevice respondsToSelector: @selector(registryID)] ? mtlDevice.registryID : 0;
 }
+
+bool mvkSupportsBCTextureCompression(id<MTLDevice> mtlDevice) {
+#if MVK_MACOS && !MVK_MACCAT
+	if ([mtlDevice respondsToSelector: @selector(supportsBCTextureCompression)]) {
+		return mtlDevice.supportsBCTextureCompression;
+	}
```

Comment:
```suggestion
#if MVK_MACOS_APPLE_SILICON
	if ([mtlDevice respondsToSelector: @selector(supportsBCTextureCompression)]) {
		return mtlDevice.supportsBCTextureCompression;
	}
#endif
```
Otherwise, the build fails on Xcode 11.

Reply:
Thanks. Yeah. CI caught this too.

I made a slightly different edit. Although in practice for now this only applies to Apple Silicon, I figured I should just address the cause of the build failure, and filter for Xcode 12 and MacCat. But please check my platform logic again.

Reply:
Do BCn textures work on Catalyst?

Reply:
>Do BCn textures work on Catalyst?

I don't know for sure. Not much on it out there and it's an inherently awkward thing to test. An iOS app that deliberately behaved so differently while running on a Mac.

I dithered on where to put Catalyst in this logic, but in the end I accepted that BCn formats are [defined for Catalyst](https://developer.apple.com/documentation/metal/mtlpixelformat/mtlpixelformatbc1_rgba/) for some reason. Since the obvious approach would be to assume an iOS app running on Catalyst wouldn't need them, I figured if Apple has gone to the trouble of declaring otherwise, they are probably supported.

Reply:
> Do BCn textures work on Catalyst?

Actually...in thinking about this further...if `supportsBCTextureCompression` is **_not_** defined for Catalyst, then an Apple Silicon macOS is not able to indicate that it doesn't support BCn. So maybe the lack of `supportsBCTextureCompression` on Catalyst is an indication that BCn are not expected to be supported on Catalyst. 

This is strange given my comment above about those formats existing on Catalyst. But maybe this is just a mixup in platform declarations in the SDK, and the prudent thing to do is assume BCn are not supported on Catalyst.

Reply:
Changed to report no BCn for Catalyst.